src/checkpoint/dispatch/dispatch_virtual.h

```diff
@@ -94,7 +107,7 @@
  *           if (!is_null) {
  *             // During size/pack, save the actual derived type of raw_ptr;
  *             // During unpack, allocate/construct raw_ptr with correct virtual type
- *             checkpoint::allocateConstructForPointer(s, raw_ptr);
+ *             checkpoint::reconstructPointedToObjectIfNeeded(s, raw_ptr);
```

Comment:
have you checked if this snippet (serializing raw pointer) actually works? IIRC the last time I've tried it with `allocateConstructForPointer` it would not compile

Reply:
I'll give it a try in the tests.

Reply:
I have checked the old version and this is working just fine:
```
template <typename T>
struct MyObjectWithRawPointer {
  T* raw_ptr;

  template <typename SerializerT>
  void serialize(SerializerT& s) {
    bool is_null = raw_ptr == nullptr;
    s | is_null; // serialize whether we have a null pointer not
    if (!is_null) {
      // During size/pack, save the actual derived type of raw_ptr;
      // During unpack, allocate/construct raw_ptr with correct virtual type
      checkpoint::allocateConstructForPointer(s, raw_ptr);
      s | *raw_ptr;
    }
  }
};

TEST_F(SomeTest, test_raw_pointer) {
  MyObjectWithRawPointer<int> obj;
  obj.raw_ptr = nullptr;

  auto ret_null = checkpoint::serialize(obj);

  int i = 3;
  obj.raw_ptr = &i;

  auto ret = checkpoint::serialize(obj);
}
```
so maybe I was just imagining things, but anyways, a test (or example?) like this could be good to have

Reply:
I just added such an example. All good?